[jormungandr] Correspondence limit by _max_additional_connections for stif

kadhikari opened the pull request:
This parameter existe also in the database of tyr: instance.max_additional_changes
For more informations : http://jira.canaltp.fr/browse/NAVP-574

Commit:
[tyr]: attribut max_additional_changes added in instance

Commit:
[tyr]: max_additional_changes added for api instances

Commit:
[tyr]: minor corrections

Commit:
[jormungandr]: max_additional_changes implemented

A test added

Commit:
[jormungandr]: minor modification

kadhikari commented:
Jenkins, test this please.

antoine-de commented on source/jormungandr/jormungandr/scenarios/journey_filter.py:
```diff
@@ -231,6 +233,49 @@ def _filter_max_successive_buses(journeys, request):
+    eliminates journeys with correspondences more then nb correspondence of best journey + _max_additional_changes
+    """
+    logger = logging.getLogger(__name__)
+    max_additional_changes = get_or_default(request, '_max_additional_changes',
```
I think `connection` is better than `changes` and `correspondences`

kadhikari replied:
Do you prefer also to change the attribt and parameter name to "max_additional_connections" and "_max_additional_connections" ?

antoine-de replied:
ask the others, but I do prefer to have connections everywhere compared to `dependences` and `changes`

antoine-de commented on source/jormungandr/jormungandr/scenarios/journey_filter.py:
```diff
@@ -231,6 +233,49 @@ def _filter_max_successive_buses(journeys, request):
+    """
+    Returns min correspondence count among journeys
+    """
+    nb_min_changes = 100
```
I think you can rewrite this like:

```
if not journeys:
return None (and check for this case in the calling method)

return min(get_nb_connections(j) for j in journeys if not _to_be_deleted(j))
```

(might not 'compil' but it's the spirit)

antoine-de commented on source/jormungandr/tests/stif_tests.py:
```diff
@@ -139,8 +140,18 @@ def test_stif_max_successive_buses_with_tram_in_between(self):
+
+        # As we modifiy the value of _max_additional_changes to 2 we delete the second journey because
+        # it contains more then nb_correspondence + 2 ()
+        query = "journeys?from={from_sp}&to={to_sp}&datetime={datetime}&_override_scenario=stif&_max_successive_buses=5"\
```
since you use the _max_additional_changes=2, I think it's better to set it in the request, not to depend on default values

antoine-de commented on source/navitiacommon/navitiacommon/default_values.py:
```diff
@@ -99,6 +99,10 @@
 bss_provider = True
 
+#Maximum number of correspondences allowed in journeys is calculated as
+#max_additional_changes + the number of correspondences of best journey
```
the comment does not seems to be up to date

Commit:
[tyr] : attrubut max_additional_changes modified to max_additional_changes

Commit:
[jormungandr]: max_additional_changes modified to max_additional_connections

some modifications on comments

kadhikari commented:
@antoine-de 
I modified max_additional_changes to max_additional_connections and replaced "correspondences", "changes" by "connections".

I also simplified the code as you suggested

Commit:
[jormungandr]: minor corrections

antoine-de commented on source/jormungandr/jormungandr/scenarios/journey_filter.py:
```diff
@@ -89,6 +89,8 @@ def final_filter_journeys(response_list, instance, request):
     if final_line_filter:
         _filter_similar_line_journeys(journeys, request)
 
+    _filter_too_much_changes(journeys, instance, request)
```
you still need to change this `changes`

Commit:
[jormungandr]: function renamed

kinnou02 commented on source/jormungandr/jormungandr/scenarios/journey_filter.py:
```diff
@@ -231,6 +233,50 @@ def _filter_max_successive_buses(journeys, request):
+    """
+    nb_connections = 0
+
+    for s in journey.sections:
```
You should use `nb_transfers` from the journey

Commit:
jormungandr: nb_transfers used to find min_changes

antoine-de commented:
Jenkins, test this please

kadhikari commented:
Jenkins, test this please
